src/core/default_appearance.js

```diff
@@ -87,6 +87,91 @@ function parseDefaultAppearance(str) {
   return new DefaultAppearanceEvaluator(str).parse();
 }
 
+class AppearanceStreamEvaluator extends EvaluatorPreprocessor {
+  constructor(stream) {
+    super(stream);
+    this.stream = stream;
+  }
+
+  parse() {
+    const operation = {
+      fn: 0,
+      args: [],
+    };
+    let result = {
+      fontSize: 0,
+      fontName: "",
+      fontColor: /* black = */ new Uint8ClampedArray(3),
+    };
+    const stack = [];
+
+    let scaleFactor = 1;
+    let mustLoop = true;
+
+    try {
+      while (mustLoop) {
+        operation.args.length = 0; // Ensure that `args` it's always reset.
+
+        if (!this.read(operation)) {
```

Comment:
To improve overall consistency, I'd probably do the following instead:
```suggestion
    let breakLoop = false;

    try {
      while (true) {
        operation.args.length = 0; // Ensure that `args` it's always reset.

        if (breakLoop || !this.read(operation)) {
```